MonoGame.Framework/Platform/Threading.cs

```diff
@@ -139,6 +139,7 @@ internal static void BlockOnUIThread<TState>(Action<TState> action, TState state
 
             try
             {
+                Debug.WriteLine("Thread blocked, waiting for action to be completed by the UI thread");
```

Comment:
These will be stripped out for release builds won't they? Which means someone using the NuGet packages will not see these messages. `Trace.WriteLine` might be a better option.

Reply:
Not sure what is the best option here, but these are our concern:

1. On release/shipping builds we don't want to slow down the game code with lots of debug checks that should not happen in a shipping game.

2. Since NuGet is our only delivery option how do we add debug/diagnostic checks that aid in development, but doesn't hurt when you ship?  Are there Debug NuGet packages?

Note this may be a bigger discussion beyond the scope of this PR.  But for games we cannot waste even a little time on debug checks that 99.99% never fire in a shipping game.

Reply:
After a quick check just there are around ~25 other Debug.WriteLine instances with potentially useful information.

Maybe they should all go to a new MonoGame method where all such output messages can be rerouted at runtime. For Debug this could always route to Debug.WriteLine as default and for Release somewhere else only if Debugger.IsAttached is true.

User defined Streams could also be attached to get these messages through a custom Console app, localhost webpage, etc.

In addition by adding unique error codes for each type of message this could be used for custom filtering and disabling of error checking/reporting by error code or error category (audio, input, graphics, video errors).

Reply:
[`Debug.WriteLine`](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Debug.cs,148) is fully optimized away ([including the interpolated string](https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQAwAIpwHQBECWAhgOYB2A9gM4Au+AxpQNzLJQDMmM6AwugN7J0QzBygAWdAFkAFAEp+g4Uvylq6AB7oAvOgCszJEqUB1AE75qAUwAyKy9IAkAIgASlgDbvy/dWF0BfJ1lGdAB6UPQAFQALfEp0dzt0OPRyAAdaAFt8AC9LYHRCAHdCAE9sRSF/SvQagG1uclJgC3wmwndpJ0iAUQBlSIBJADkAcSCAXTCI4fI1YEsAMzsCwlJSoujLU0sKw2E08wA3QitMCTMLGztpLAxKeRqBJABifEX0XB6AIQBVUZqwiwAE5pA8DC9LM13koatUkP4gA===)) in non-DEBUG builds.

I added a [`GraphicsDebug.Blah`](https://github.com/Mindfulplays/MonoGame/blob/ios-metal/MonoGame.Framework/Platform/Graphics/GraphicsDebug.Default.cs#L115) in my private repo to streamline these. Maybe I can bring these in a separate PR?

Reply:
>     1. On release/shipping builds we don't want to slow down the game code with lots of debug checks that should not happen in a shipping game.

Use Assert instead of "try catch".